thetis/callback.py

```diff
@@ -161,7 +161,9 @@ def __init__(self, solver_obj, array_dim=1, attrs=None,
                  export_to_hdf5=True,
                  append_to_log=True,
                  include_time=True,
-                 hdf5_dtype='d'):
+                 hdf5_dtype='d',
+                 start_time=-numpy.inf,
+                 end_time=numpy.inf):
```

Comment:
I'd prefer that the default unspecified case is `None` in the API so you don't need to carry around `numpy.inf` as the default value in the derived classes. Internally setting to inf is fine.